Pod/Classes/Categories/UIImage+DecodedImage.m

```diff
@@ -69,22 +83,91 @@ + (UIImage *)pin_decodedImageWithCGImageRef:(CGImageRef)imageRef
     }
     
     CGSize imageSize = CGSizeMake(CGImageGetWidth(imageRef), CGImageGetHeight(imageRef));
-    UIGraphicsBeginImageContextWithOptions(imageSize, opaque, 1.0);
-    CGContextRef ctx = UIGraphicsGetCurrentContext();
+    
+    CGBitmapInfo info = opaque ? (kCGImageAlphaNoneSkipFirst | kCGBitmapByteOrder32Host) : (kCGImageAlphaPremultipliedFirst | kCGBitmapByteOrder32Host);
+    CGColorSpaceRef colorspace = CGColorSpaceCreateDeviceRGB();
+    
+    //Use UIGraphicsBeginImageContext parameters from docs: https://developer.apple.com/library/ios/documentation/UIKit/Reference/UIKitFunctionReference/#//apple_ref/c/func/UIGraphicsBeginImageContextWithOptions
+    CGContextRef ctx = CGBitmapContextCreate(NULL, imageSize.width, imageSize.height,
+                                             8,
+                                             0,
+                                             colorspace,
+                                             info);
+    
+    CGColorSpaceRelease(colorspace);
+    
```

Comment:
That all makes great sense.

I asked the question (well, first, because I didn't understand why this was necessary, now I do) because I have a vague memory that creating bitmaps of width not being some multiple make the GPU unhappy (and that using UIGraphicsBeginImageContextWithOptions took care of it for you). But I can't find any reference / documentation around this. I'll dig some more, but this seems like the correct direction.